### ISSUE TITLE
[17.0] Geoip not working correctly

### ISSUE THREAD

**Issue by shaneo1** [Tecnativa/doodba#600]
I have installed Odoo 17 using Doodba and use global price lists and have used this in odoo 16 with the geolite2 database. But was only able to get it working using a thirdparty app from odoo.com. 

I have installed odoo 17 using a different docker-compose script that puts both Geolite2-city and country in to /usr/share/GeoIP as instructed my the odoo documentation.

Now in Doodba there is on the city database and the odoo.conf in pointing to its location geoip_database = /opt/odoo/auto/geoip/GeoLite2-City.mmdb. and the geoip does not work for the country,  I believe this is a bug in the scripting as I am unable to detect the customer country against the IP address. How can I rectify this issue.

**Comment by ap-wtioit** [comment]
@shaneo1 did you setup the geoip credentials correctly? (is there a database in /opt/odoo/auto/geoip/GeoLite2-City.mmdb?)

**Comment by shaneo1** [comment]
There is a city database yes.

**Comment by shaneo1** [comment]
> Ok just investigated (because we have an issue with the [new download limits for GeoIP](https://dev.maxmind.com/geoip/release-notes/2024#api-policies---temporary-enforcement-on-february-7-2024) and testing on our CI) and i can confirm the parameters for GeoIP in Odoo have changed in Odoo 17.0 and are no longer compatible with the doodba implementation:
> 
> [Odoo 17.0](https://github.com/odoo/odoo/blob/17.0/odoo/tools/config.py#L318)
> 
> ```
>         group.add_option("--geoip-city-db", "--geoip-db", dest="geoip_city_db", my_default='/usr/share/GeoIP/GeoLite2-City.mmdb',
>                          help="Absolute path to the GeoIP City database file.")
>         group.add_option("--geoip-country-db", dest="geoip_country_db", my_default='/usr/share/GeoIP/GeoLite2-Country.mmdb',
>                          help="Absolute path to the GeoIP Country database file.")
> ```
> 
> [Odoo 16.0 and before](https://github.com/odoo/odoo/blob/16.0/odoo/tools/config.py#L319)
> 
> ```
>          group.add_option("--geoip-db", dest="geoip_database", my_default='/usr/share/GeoIP/GeoLite2-City.mmdb',
>                          help="Absolute path to the GeoIP database file.")
> ```
> 
> As a workaround you should be able to add the following file to your odoo/custom/conf.d directory: `994_geoip.conf`
> 
> ```
> [options]
> geoip_city_db = /opt/odoo/auto/geoip/GeoLite2-City.mmdb
> ```
> 
> @shaneo1 could you update the issue title with a prefix `[17.0] ` to make it clear to everyone that this is for 17.0

Thank you for you reply I didn't think I was going crazy when I submitted the issue. As for the file 994_geoip.conf where can I find this?

**Comment by ap-wtioit** [comment]
The file should be a text file with the contents above and should be placed inside the odoo/custom/conf.d directory of your doodba project. You can have a look at the tests for custom settings in the [settings test](https://github.com/Tecnativa/doodba/tree/master/tests/scaffoldings/settings/custom/conf.d). All files in the conf.d directory should be merged together into /opt/odoo/auto/odoo.conf in the container.

**Comment by shaneo1** [comment]
Ok now have this working in Odoo 17 here are the changes I made the project to get GeoLite2 to place nicely. 

```
volumes:
      - ./odoo/GeoLite2:/usr/share/GeoIP:rw
      - ./odoo/custom/conf.d/geoip.conf:/etc/odoo/odoo.conf.d/geoip.conf:ro
    command:
      - --geoip-city-db=/usr/share/GeoIP/GeoLite2-City.mmdb
      - --geoip-country-db=/usr/share/GeoIP/GeoLite2-Country.mmdb
      - --config=/etc/odoo/odoo.conf.d/geoip.conf
```
The folder /odoo/GeoLite2 was added and both GeoLite2-City.mmdb and GeoLite2-Country.mmdb uploaded from Maxmind website. In addition I created a geoip.conf file in /odoo/custom/conf.d

```
[options]
geoip_database = /usr/share/GeoIP/GeoLite2-City.mmdb
```
Then run docker-compose up -d 

Everything is now working as it should be.

**Comment by pedrobaeza** [comment]
Fixed in #603

**Comment by shaneo1** [comment]
Awesome news this will help a lot of people out. Great work everyone
